### PR TITLE
Remove prompt pulse

### PR DESCRIPTION
**Pull request by dmitry-ganyushin**
### Description of work
[6177: RemovePromptPulse doesn't work when negative time-of-flight are in the workspace](https://ornlrse.clm.ibmcloud.com/ccm/resource/itemName/com.ibm.team.workitem.WorkItem/6177)

#### Summary of work
This PR should fix issues for [RemovePromptPulse](https://docs.mantidproject.org/nightly/algorithms/RemovePromptPulse-v1.html)  to mask correctly the first pulse including the case of ToFmin <0.

### CODE REVIEW

**Comment by sf1919** [comment]
Don't forget to add labels and milestones please

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
This looks like it fixed the bug, but it needs a release note

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
1. Still needs release notes
2. I prefer you verify visually that only low TOF/d-spacing is changed and update the reference file ([docs for updating files](https://developer.mantidproject.org/DataFilesForTesting.html#updating-file-s))

**Comment by dmitry-ganyushin** [comment]
If I set ```Width=1```, the produced ```PG3_4866.gsa``` file is bit identical to the golden one. This test is Ok on my local installation; I am trying to figure out what is wrong here.

**Comment by peterfpeterson** [comment]
It looks like you are only running a single system test. Run the full SNSPowderRedux suite locally and you'll track down the changes faster.
```sh
ninja && ./systemtest -R SNSPowderRedux
```

Also, you need to put the input parameters in the tests back to what they were. This is how the instrument team has been calling the code for years and it needs to stay a realistic example. The reference data needs to be inspected to see how it is changed and [update the reference file](https://developer.mantidproject.org/DataFilesForTesting.html#updating-file-s).

File: Testing/SystemTests/tests/framework/SNSPowderRedux.py
Comment (peterfpeterson) [inline Testing/SystemTests/tests/framework/SNSPowderRedux.py:88]: Please undo the changes to the test inputs. The entirety of `SNSPowderRedux.py` needs to go back to the way it was

**Review by peterfpeterson — DISMISSED** [review]

**Comment by sf1919** [comment]
This is blocked by #37972. The Linux tests here can be re-run once it has been merged in.

**Comment by dmitry-ganyushin** [comment]
Closed in favor of https://github.com/mantidproject/mantid/pull/37982. Could not get this PR get though the build, probably because it started from ```main```. https://github.com/mantidproject/mantid/pull/37982 starts form ```release-next```

**Pull request closed without merging**
